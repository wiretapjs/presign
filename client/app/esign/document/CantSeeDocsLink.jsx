import React from 'react'
import Dialog from 'material-ui/Dialog'
import PropTypes from 'prop-types'
import { StyledLink } from './HelpLink.style'
import { StyledCantseeLinkWrapper } from './CantSeeDocsLink.style'
import DialogActions from './DialogActions'

const CantSeeDocsLink = props => {
  return (
    <StyledCantseeLinkWrapper>
      <StyledLink onClick={props.openDialog}>Can't see your documents?</StyledLink>
      <Dialog
        title="Check your browser settings"
        actions={[<DialogActions onClick={props.closeDialog} />]}
        modal={false}
        contentStyle={{ borderTop: `5px solid #ECAC00` }}
        open={props.isOpen}
        onRequestClose={props.closeDialog}
      >
        <p>Please update your settings to:</p>
        <ol>
          <li>Always allow cookies</li>
          <li>Turn off any ad blockers or add-ons</li>
        </ol>
        <p>Once your settings are updated, refresh your browser for the documents to load.</p>
      </Dialog>
    </StyledCantseeLinkWrapper>
  )
}

CantSeeDocsLink.propTypes = {
  isOpen: PropTypes.bool,
  openDialog: PropTypes.func,
  closeDialog: PropTypes.func,
}

export default CantSeeDocsLink
