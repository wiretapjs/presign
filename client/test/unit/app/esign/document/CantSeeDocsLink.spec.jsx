import React from 'react'
import { shallow } from 'enzyme'
import CantSeeDocsLink from 'app/esign/document/CantSeeDocsLink'

describe('CantSeeDocsLink', () => {
  it('toggles openDialog after clicking on StyledLink', () => {
    const CantSeeDocsLinkWrapper = shallow(<CantSeeDocsLink />)
    expect(CantSeeDocsLinkWrapper).toMatchSnapshot()
  })
})
