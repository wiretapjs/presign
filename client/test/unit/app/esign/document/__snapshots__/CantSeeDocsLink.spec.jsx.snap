// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CantSeeDocsLink toggles openDialog after clicking on StyledLink 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <styled.div
      onClick={undefined}
    >
      Can't see your documents?
    </styled.div>
    <Dialog
      actions={
        Array [
          <DialogActions
            onClick={undefined}
          />,
        ]
      }
      autoDetectWindowHeight={true}
      autoScrollBodyContent={false}
      contentStyle={
        Object {
          "borderTop": "5px solid #ECAC00",
        }
      }
      modal={false}
      onRequestClose={undefined}
      open={undefined}
      repositionOnUpdate={true}
      title="Check your browser settings"
    >
      <p>
        Please update your settings to:
      </p>
      <ol>
        <li>
          Always allow cookies
        </li>
        <li>
          Turn off any ad blockers or add-ons
        </li>
      </ol>
      <p>
        Once your settings are updated, refresh your browser for the documents to load.
      </p>
    </Dialog>
  </styled.div>,
  "nodes": Array [
    <styled.div>
      <styled.div
        onClick={undefined}
      >
        Can't see your documents?
      </styled.div>
      <Dialog
        actions={
          Array [
            <DialogActions
              onClick={undefined}
            />,
          ]
        }
        autoDetectWindowHeight={true}
        autoScrollBodyContent={false}
        contentStyle={
          Object {
            "borderTop": "5px solid #ECAC00",
          }
        }
        modal={false}
        onRequestClose={undefined}
        open={undefined}
        repositionOnUpdate={true}
        title="Check your browser settings"
      >
        <p>
          Please update your settings to:
        </p>
        <ol>
          <li>
            Always allow cookies
          </li>
          <li>
            Turn off any ad blockers or add-ons
          </li>
        </ol>
        <p>
          Once your settings are updated, refresh your browser for the documents to load.
        </p>
      </Dialog>
    </styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <CantSeeDocsLink />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <styled.div
            onClick={undefined}
          >
            Can't see your documents?
          </styled.div>
          <Dialog
            actions={
              Array [
                <DialogActions
                  onClick={undefined}
                />,
              ]
            }
            autoDetectWindowHeight={true}
            autoScrollBodyContent={false}
            contentStyle={
              Object {
                "borderTop": "5px solid #ECAC00",
              }
            }
            modal={false}
            onRequestClose={undefined}
            open={undefined}
            repositionOnUpdate={true}
            title="Check your browser settings"
          >
            <p>
              Please update your settings to:
            </p>
            <ol>
              <li>
                Always allow cookies
              </li>
              <li>
                Turn off any ad blockers or add-ons
              </li>
            </ol>
            <p>
              Once your settings are updated, refresh your browser for the documents to load.
            </p>
          </Dialog>
        </styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <styled.div
            onClick={undefined}
          >
            Can't see your documents?
          </styled.div>
          <Dialog
            actions={
              Array [
                <DialogActions
                  onClick={undefined}
                />,
              ]
            }
            autoDetectWindowHeight={true}
            autoScrollBodyContent={false}
            contentStyle={
              Object {
                "borderTop": "5px solid #ECAC00",
              }
            }
            modal={false}
            onRequestClose={undefined}
            open={undefined}
            repositionOnUpdate={true}
            title="Check your browser settings"
          >
            <p>
              Please update your settings to:
            </p>
            <ol>
              <li>
                Always allow cookies
              </li>
              <li>
                Turn off any ad blockers or add-ons
              </li>
            </ol>
            <p>
              Once your settings are updated, refresh your browser for the documents to load.
            </p>
          </Dialog>
        </styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CantSeeDocsLink />,
}
`;
